convertercurrency/src/main/java/page/devnet/convertercurrency/fxratesapi/FxRatesApiService.java

```diff
@@ -0,0 +1,122 @@
+package page.devnet.convertercurrency.fxratesapi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import page.devnet.convertercurrency.ConverterCurrencyException;
+import page.devnet.convertercurrency.ConverterCurrencyService;
+import page.devnet.convertercurrency.fxratesapi.pojo.FxRatesApiResponse;
+import page.devnet.convertercurrency.utils.ParserCurrencyMessage;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Konstantin Agafonov
+ * @since 01.08.24
+ */
+@Slf4j
+public class FxRatesApiService implements ConverterCurrencyService {
+
+    private final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+    private final URI appCurrencyApiUrl;
+    private final HttpClient client;
+    private final ParserCurrencyMessage parserCurrencyMessage = new ParserCurrencyMessage();
+    private static final List<String> currencies = Arrays.asList("RUB", "USD", "EUR", "JPY", "KZT", "GEL", "NZD");
```

Comment:
i don't really understood why we double this list. Can we exctract it and reuse in all places?

Maybe something like CurrencyDictionary.listCurrencies()